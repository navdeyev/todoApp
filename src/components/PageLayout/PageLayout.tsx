import * as React from 'react';

import PageHeader from 'components/PageHeader/PageHeader';
import PageTitle from 'components/PageTitle/PageTitle';
import styled from 'components/Styled/styledComponents';
import {rem} from 'components/Styled/utils';

const Page = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
`;
Page.displayName = 'Page';

const PageContentHolder = styled.div`
    flex: 1 0 0px;
    overflow-y: scroll;
    padding-top: ${ rem(16) };
`;
PageContentHolder.displayName = 'PageContentHolder';

export interface IProps {
    children: React.ReactChild;
}

const PageLayout: React.SFC<IProps> = (props) => {
    const {children} = props;
    return (
        <Page>
            <PageHeader>
                <PageTitle>Welcome to TodoApp</PageTitle>
            </PageHeader>
            <PageContentHolder>{children}</PageContentHolder>
        </Page>
    );
};

export default PageLayout;
