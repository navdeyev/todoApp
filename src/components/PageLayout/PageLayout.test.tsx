import {shallow} from 'enzyme';
import * as React from 'react';

import {IProps, PageLayout} from './PageLayout';

describe('PageLayout', () => {

    it('renders a component', () => {
        const defaultProps: IProps = {
            changeTheme: jest.fn(),
            children: <div>Children</div>
        };
        const layout = shallow(<PageLayout {...defaultProps} />);
        expect(layout.getElement()).toMatchSnapshot();
    });

});
