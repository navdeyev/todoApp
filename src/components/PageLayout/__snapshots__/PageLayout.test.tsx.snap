// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders a component 1`] = `
<ThemeProvider
  theme={[ThemeObject]}
>
  <Page
    theme={[ThemeObject]}
  >
    <PageHeader
      theme={[ThemeObject]}
    >
      <PageTitle
        theme={[ThemeObject]}
      >
        Welcome to TodoApp
      </PageTitle>
      <ThemeSelector
        changeTheme={[MockFunction]}
        theme={[ThemeObject]}
      />
    </PageHeader>
    <PageContentHolder>
      <div>
        Children
      </div>
    </PageContentHolder>
  </Page>
</ThemeProvider>
`;
